adot-testbed/app/src/test/java/software/amazon/adot/testbed/LogsTests.java

```diff
@@ -65,61 +73,167 @@ private GenericContainer<?> createAndStartCollector(String configFilePath, Strin
+       //Mount the Temp directory
+        collector.withFileSystemBind(logDirectory.toString(),"/tempLogs", BindMode.READ_WRITE);
 
         collector.start();
-        collector.waitingFor(Wait.forHealthcheck());
         return collector;
     }
 
     @Test
     void testSyslog() throws Exception {
         String logStreamName = "rfcsyslog-logstream-" + uniqueID;
-        collector = createAndStartCollector("/configurations/config-rfcsyslog.yaml", "/logs/RFC5424.log", logStreamName);
+        collector = createAndStartCollector("/configurations/config-rfcsyslog.yaml", logStreamName);
 
-        validateLogs(logStreamName , "/logs/RFC5424.log");
+        List<String> logFilePaths = new ArrayList<>();
+        logFilePaths.add("/logs/RFC5424.log");
+        validateLogs(logStreamName , logFilePaths, true);
         collector.stop();
     }
 
     @Test
     void testLog4j() throws Exception {
         String logStreamName = "log4j-logstream-" + uniqueID;
-        collector = createAndStartCollector("/configurations/config-log4j.yaml", "/logs/log4j.log", logStreamName);
+        collector = createAndStartCollector("/configurations/config-log4j.yaml", logStreamName);
 
-        validateLogs(logStreamName , "/logs/log4j.log");
+        List<String> logFilePaths = new ArrayList<>();
+        logFilePaths.add("/logs/log4j.log");
+        validateLogs(logStreamName , logFilePaths, true);
         collector.stop();
     }
 
     @Test
     void testJson() throws Exception {
         String logStreamName = "json-logstream-" + uniqueID;
-        collector = createAndStartCollector("/configurations/config-json.yaml", "/logs/testingJSON.log", logStreamName);
+        collector = createAndStartCollector("/configurations/config-json.yaml", logStreamName);
 
-        validateLogs(logStreamName , "/logs/testingJSON.log");
+        List<String> logFilePaths = new ArrayList<>();
+        logFilePaths.add("/logs/testingJSON.log");
+        validateLogs(logStreamName , logFilePaths, true);
         collector.stop();
     }
 
-    void validateLogs(String testLogStreamName, String logFilePath) throws Exception {
-        var file = new File(logFilePath);
+    @Test
+    void testCollectorRestartStorageExtension() throws Exception {
+        String logStreamName = "storageExtension-logstream-" + uniqueID;
+        collector = createAndStartCollector("/configurations/config-storageExtension.yaml", logStreamName);
+        File tempFile = new File(logDirectory.toString(), "storageExtension.log");
+        Thread.sleep(5000);
+
+        FileWriter fileWriter = new FileWriter(tempFile);
+        try {
+            fileWriter.write("First Message, collector is running" + "\n");
+            fileWriter.write("Second Message, collector is running" + "\n");
+            fileWriter.write("Third Message,  collector is running" + "\n");
+        } catch (IOException e) {
```

Comment:
do you need this since your method has a `throws Exception`? to simplify the code, I think you don't need to catch this exception here.

Reply:
Simplified as suggested.